nixos/modules/services/audio/mpdscheduler.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let 
+  cfg = config.services.mpdscheduler;
+in {
+  options.services.mpdscheduler = {
+    enable = mkEnableOption "mpdscheduler service";
+    host = mkOption {
+      type = types.str;
+      description = "the mpd host to connect to";
+      default = "localhost";
+      example = "mpdHost.local";
+    };
+    port = mkOption {
+      type = types.int;
+      description = "the mpd tcp port to connect to";
+      default = 6600;
+    };
+    additionalChannels = mkOption {
+      type = types.bool;
+      default = true;
+      description = "whether the 'alarm', 'sleep' and 'cancel' channels shall be used in addition to the 'scheduler' channel";
+    };
+    fadeTime = mkOption {
+      type = types.int;
+      default = 30;
+      description = "alarm/sleep fade time in seconds";
+      example = 15;
+    };
+    maxVolume = mkOption {
```

Comment:
Should we validate this is between 1 and 100?

Reply:
technically, you could wake at 0% as well, so i would prefer between 0 and 100.

How can you specify a range? (This is my first module)

Reply:
> types.ints.between lowest highest
    An integer between lowest and highest (both inclusive). Useful for creating types like types.port. 

See NixOS manual for reference.

Reply:
Thanks. I also changed the fade time to be non-negative.